plugins/inputs/whois/whois.go

```diff
@@ -0,0 +1,223 @@
+package whois
+
+import (
+	_ "embed"
+	"errors"
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/likexian/whois"
+	"github.com/likexian/whois-parser"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+type Whois struct {
+	Domains            []string        `toml:"domains"`
+	Server             string          `toml:"server"`
+	Timeout            config.Duration `toml:"timeout"`
+	IncludeNameServers bool            `toml:"include_name_servers"`
+	Log                telegraf.Logger `toml:"-"`
+
+	Client         *whois.Client
+	WhoisLookup    func(client *whois.Client, domain, server string) (string, error)
+	ParseWhoisData func(raw string) (whoisparser.WhoisInfo, error)
```

Comment:
Doesn't need to be exported
```suggestion
	client         *whois.Client
	whoisLookup    func(client *whois.Client, domain, server string) (string, error)
	parseWhoisData func(raw string) (whoisparser.WhoisInfo, error)
```

Reply:
Fixed